crypto/camellia/camellia.c

```diff
@@ -57,25 +57,16 @@
 #  define GETU32(p)   SWAP(*((u32 *)(p)))
 #  define PUTU32(p,v) (*((u32 *)(p)) = SWAP((v)))
 # elif defined(__GNUC__) && __GNUC__>=2
-#  if defined(__i386) || defined(__x86_64)
-#   define RightRotate(x,s) ({u32 ret; asm ("rorl %1,%0":"=r"(ret):"I"(s),"0"(x):"cc"); ret; })
-#   define LeftRotate(x,s)  ({u32 ret; asm ("roll %1,%0":"=r"(ret):"I"(s),"0"(x):"cc"); ret; })
-#   if defined(B_ENDIAN)        /* stratus.com does it */
-#    define GETU32(p)   (*(u32 *)(p))
-#    define PUTU32(p,v) (*(u32 *)(p)=(v))
-#   else
-#    define GETU32(p)   ({u32 r=*(const u32 *)(p); asm("bswapl %0":"=r"(r):"0"(r)); r; })
-#    define PUTU32(p,v) ({u32 r=(v); asm("bswapl %0":"=r"(r):"0"(r)); *(u32 *)(p)=r; })
-#   endif
```

Comment:
Not that it matters (there is assembly implementation and it's not exactly critical algorithm), but why? Or if we take this path, then why not remove even below segments for consistency sake?

Reply:
On one side it uses unaligned p, so I should rewrite it to use fast_u32 like below.
But on the other side, it is impossible to compile this section because camellia.c
is only compiled with no-asm, but then this whole section is excluded because
OPENSSL_NO_ASM is defined.
So unless I am mistaken this was already dead code for a long time.

Reply:
There is hint above where it's compiled, at stratus.com's VOS. Yes, it's x86 and big-endian. It's not a joke or typo. They solve it in compiler. But as already said it's not critical, so just remove whole `__GNUC__` segment. Or why not whole block, i.e. one starting with !defined(PEDANTIC). It can be noted that contemporary compilers can recognize shifts and or and generate rotate instruction, so it's only byte swaps and misaligned access one misses...

Reply:
what is the Configure line how this is invoked, or are they by-passing our make rules?

Reply:
vos-gcc target.

Reply:
It's not a problem to check s390x code. Everybody can do it with s390x cross-compiler and qemu-s390x. Both are readily available as pre-compiled options for Ubuntu and Debian. And I do have access to hardware and run periodic tests.

Reply:
I could even fix the x86 section if you could test it on a vos target system?

Reply:
> test it on a vos

I can't. As already said, it's not critical algorithm, no big loss, just omit...

Reply:
`./Configure linux-generic32` invokes the removed code section, sorry for not missing that.